lib/cartopy/tests/test_polygon.py

```diff
@@ -446,6 +446,18 @@ def test_inverted_poly_simple_hole(self):
         self._assert_bounds(polygon.interiors[0].bounds,
                             - 1.2e7, -1.2e7, 1.2e7, 1.2e7, 1e6)
 
+    def test_inverted_poly_multi_hole(self):
+        # Adapted from 1149
+        proj = ccrs.LambertAzimuthalEqualArea(
+            central_latitude=45, central_longitude=-100)
+        poly = sgeom.Polygon([(-180, -80), (180, -80), (180, 90), (-180, 90)],
+                             [[(-50, -50), (-50, 0), (0, 0), (0, -50)]])
+        multi_polygon = proj.project_geometry(poly)
+        # Should project to single polygon with multiple holes
+        assert len(multi_polygon.geoms) == 1
+        assert len(multi_polygon.geoms[0].interiors) >= 2
```

Comment:
We actually get 36 holes.  I assume most of them are responsible for the seam down the dateline visible in https://github.com/SciTools/cartopy/pull/2470#issuecomment-2453463888

Against main, this test fails at the first `assert` since we have two polygons in our multipolygon.

Reply:
I would remove the image test since it is downloading the natural earth ocean for every runner, or replace it with this example as an image comparison test that doesn't require external downloading.

The 36 holes is interesting and worth investigating IMO. For your example here, it looks like there are many small loops around the boundary points that are created. (your image has a small white curve/line in it protruding from the south pole area which are these linear rings I think)
```python
import matplotlib.pyplot as plt

ax = plt.figure().add_subplot()

interior = multi_polygon.geoms[0].interiors
for i in range(len(interior)):
    ax.plot(*interior[i].xy)

plt.show()
```

I wonder if these extra loops/rings are what is slowing down some of our projecting of geometries even 🤔 

I'm also happy to defer these investigations to future PRs because this is already a pretty substantial improvement I think.